### PR TITLE
Reuse client factory for rest and handler usage

### PR DESCRIPTION
**Pull request by patduin**
Alternative to https://github.com/HotelsDotCom/waggle-dance/pull/37 if we don't like the pool.

Update:
Leaving the pool.
Going for this solution now. Fixes issue #34
Please review and I'll get on a release.

### CODE REVIEW

**Comment by patduin** [comment]
@massdosage yeah there's still quite a bit of refactoring to be done having slept on it they won't be that far off tbh. But yes could do it in two steps.

**Comment by ddcprg** [comment]
I've not fully looked at the other PR but this looks neater. I'd would also go for this change first and then we can see how we can make this more efficiently.

File: waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/SimpleFederationStatusService.java [hunk 1]
```diff
@@ -44,17 +46,13 @@ private HiveConf newHiveConf(String metaStoreUris) {
...
-  public MetaStoreStatus checkStatus(String metaStoreUris) {
-    HiveConf hiveConf = newHiveConf(metaStoreUris);
-    HiveMetaStoreClient client = null;
-    try {
-      client = new HiveMetaStoreClient(hiveConf);
+  public MetaStoreStatus checkStatus(AbstractMetaStore abstractMetaStore) {
+    try (CloseableThriftHiveMetastoreIface client = metaStoreClientFactory.newInstance(abstractMetaStore)) {
+      if (!client.isOpen()) {
+        return MetaStoreStatus.UNAVAILABLE;
+      }
```
Comment (massdosage) [inline waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/SimpleFederationStatusService.java]: Where does the closing of the client happen now?

File: waggle-dance-rpm/pom.xml [hunk 2]
```diff
@@ -26,7 +26,7 @@
         <artifactId>maven-dependency-plugin</artifactId>
         <executions>
           <execution>
-            <id>circus-train-bin</id>
+            <id>waggle-dance-bin</id>
```
Comment (massdosage) [inline waggle-dance-rpm/pom.xml]: Heh, good spot.

File: waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/SimpleFederationStatusService.java (hunk 1 above)
Comment (patduin) [reply waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/SimpleFederationStatusService.java]: Try with resources, java does it for me \o/

Comment (massdosage) [reply waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/SimpleFederationStatusService.java]: Aah, of course!

**Pull request merged**